HumanControls/HumanControls.ino

```diff
@@ -56,8 +78,76 @@ void onPinActivated(int pinNr)
     switch (pinNr)
     {
     case ENCODER_PIN_SW:
-        menuController.menuPress();
+        menuController.menuPress(status, statuses, fireController);
+        break;
+    case ENABLE_PIN:
+        enableController.setIsEnabled(true);
+        break;
+    case FIRE_PIN:
+        //Eventually will use to tell Robot to fire through JSON
+        if (status == statuses[2] && isConnected)
+        {
+            fireController.initiateFiring(true);
+            fireController.setIsFireToggled(false);
+        }
+        break;
+    }
+}
+
+void onPinDeactivated(int pinNr)
+{
+    switch (pinNr)
+    {
+    case ENABLE_PIN:
+        enableController.setIsEnabled(false);
+        fireController.setIsFireToggled(false);
         break;
     }
 }
-void onPinDeactivated(int pinNr) {}
\ No newline at end of file
+
+void setStatus()
+{
+
+    if (isConnected)
+    {
+        //Check the robot's status and see if it is different then the controllers.
+        //If it is, set the status to the lowest given status. Ex. Robot returns disabled, but controller enabled, so
+        //set both statuses to disabled. Robot status and controller status should always be the same.
+        //Only exception may be due to firing setting status to enabled after firing.
+
+        if (enableController.getIsEnabled())
+        {
+            if (fireController.getIsFireToggled())
+            {
+                status = statuses[2];
+            }
+            else
+            {
+                status = statuses[1];
+            }
+        }
+        else
+        {
+            status = statuses[0];
+        }
+    }
+    else
+    {
+        status = "disconeccted";
+    }
+
+    if (status != lastStatus)
+    {
+        Serial.println(status);
+        lastStatus = status;
+        menuController.menuRefresh(status);
+    }
+
+    //Enventually will set the robot's status here
+}
+
+void connect()
+{
+    //Connect to the robot
+    isConnected = true;
+}
```

Comment:
I'd like to try to move most of the logic outside of the .ino file into classes.

Reply:
Done.